src/QGONexum.py

```diff
@@ -159,61 +198,43 @@ def combineNodeCoords(node_coords_1, node_coords_2):
     return {**node_coords_1, **node_coords_2}
 
 
-class QAOANode:
-
-    neighbors = []
-
-    def __init__(self, point, name):
-        self.point = point
-        self.coordinates = ()
-        try:
-            self.coordinates = (point["geometry"].x, point["geometry"].y)
-        except: 
-            self.coordinates = (point["geometry"].centroid.x, point["geometry"].centroid.y)
-            
-        self.name = name
-    
-    def setCoordinates(self, coordinates):
-        self.coordinates = coordinates
-
-    def getCoordinates(self):
-        return self.coordinates
-    
-    def changeName(self, new_name):
-        self.name = new_name
-        
-    def __str__(self):
-        return self.name
-
-
 class QGOProblem:
 
     #GeoJSON Files
-    files = []
+    files: list[gpd.GeoDataFrame] = []
 
     #Node Lists for each GeoJSON File
-    nodes_lists = []
+    nodes_lists: list[list[QAOANode]] = []
 
     #Coordinate Lists for each GeoJSON File
-    coords_lists = []
+    coords_lists: list[tuple[float, float]] = []
 
     #Node Coord Dictionaries for each GeoJSON File
-    ncds = []
+    ncds: list[dict[QAOANode, tuple[float, float]]] = []
 
     #Prefixes for the names of nodes for each GeoJSON File
-    prefixes = []
+    prefixes: list[str] = []
 
+    def __init__(self, links: list[str], prefixes: list[str], cutoffs: list[str]):
 
+        self.setFiles(links)
+        self.prefixes = prefixes
+        
+        self.setNodeLists(cutoffs, prefixes)
+        self.setCoordsLists()
+        self.setNodeCoordDictionaries()
```

Comment:
Moved the constructor to the top of method list.

Reply:
Thank you!